packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -179,20 +179,24 @@ export function buildExtraViteConfig(
 	options: ResolvedOptions,
 	config: UserConfig
 ): Partial<UserConfig> {
-	const allSvelteImports = [...SVELTE_IMPORTS, ...SVELTE_HMR_IMPORTS];
-
-	// exclude svelte imports from optimization unless explicitly included
-	const excludeFromOptimize = allSvelteImports.filter(
-		(x) => !config.optimizeDeps?.include?.includes(x)
-	);
-
+	// include svelte imports for optimization unless explicitly excluded
+	const include: string[] = [];
+	const exclude: string[] = ['svelte-hmr'];
```

Comment:
Do we want to add all of `SVELTE_HMR_IMPORTS` here? Or does this already work fine :thinking:

Reply:
Answer from dominik on discord: Vite is able to detect bare package name: https://github.com/vitejs/vite/blob/5c0f0a362f669e3d38ddc51bfdd9d424a7c08feb/packages/vite/src/node/optimizer/scan.ts#L272

Reply:
svelte-hmr in there covers svelte-hmr and all imports starting with svelte-hmr/ so that should work.